scripts/styleChecker.fsx

```diff
@@ -0,0 +1,393 @@
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierInstalled) then
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"install prettier@{version}"
+            },
+            Echo.Off
+        )
+        |> UnwrapProcessResult "" true
+        |> ignore
+
+let StyleFSharpFiles(rootDir: DirectoryInfo) =
+    InstallFantomlessTool(fantomlessToolVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "dotnet"
+                Arguments = $"fantomless --recurse {rootDir.FullName}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let StyleCSharpFiles(rootDir: DirectoryInfo) =
+    Process
+        .Execute(
+            {
+                Command = "dotnet"
+                Arguments = $"format whitespace {rootDir.FullName} --folder"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let InstallPrettierPluginXml(version: string) =
+    let isPrettierPluginXmlInstalled =
+        Process.Execute(
+            {
+                Command = "npm"
+                Arguments = $"list @prettier/plugin-xml@{version}"
+            },
+            Echo.Off
+        )
+        |> IsProcessSuccessful
+
+    if not(isPrettierPluginXmlInstalled) then
+        Process
+            .Execute(
+                {
+                    Command = "npm"
+                    Arguments = $"install @prettier/plugin-xml@{version}"
+                },
+                Echo.Off
+            )
+            .UnwrapDefault()
+        |> ignore
+
+let StyleXamlFiles() =
+    InstallPrettier(prettierVersion)
+    InstallPrettierPluginXml(pluginXmlVersion)
+
+    Process
+        .Execute(
+            {
+                Command = "npm"
+                Arguments =
+                    $"install --save-dev prettier@{prettierVersion} @prettier/plugin-xml@{pluginXmlVersion}"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+    Process
+        .Execute(
+            {
+                Command = "./node_modules/.bin/prettier"
+                Arguments =
+                    "--xml-whitespace-sensitivity ignore --tab-width 4 --prose-wrap preserve --write '**/*.xaml'"
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+let RunPrettier(arguments: string) =
+
+    // We need this step so we can change the files using `npx prettier --write` in the next step.
+    // Otherwise we get permission denied error in the CI.
+    Process
+        .Execute(
+            {
+                Command = "chmod"
+                Arguments = "777 -R ."
+            },
+            Echo.Off
+        )
+        .UnwrapDefault()
+    |> ignore
+
+    Process.Execute(
+        {
+            Command = "npx"
+            Arguments = $"prettier {arguments}"
+        },
+        Echo.Off
+    )
+    |> UnwrapProcessResult "" true
```

Comment:
@realmarv when you are calling UnwrapProcessResult with `""` and `true`, does it act differently than Process.UnwrapResult function from Fsdk?

Reply:
when the process result type is WarningsOrAmbiguous, ProcessResult.Unwrap function raises an error but we don't want to raise that error because if you remember, npm had an unexpected output (I guess it was writing its output in stderr even when there was no error or something)

Reply:
@realmarv so, you're saying that when passing `true` to UnwrapProcessResult (which means that the param `raiseError` = true), then you DO NOT raise an error (if warning)? If the answer to this is yes, do you understand why this is extremely confusing to me?

Reply:
Yes, that's the case. We never want to raise an error in case of a warning and we want to decide whether to raise an error or not in case of an error.

Reply:
then I'd say it's easier to read if you rename the param to `ignoreErrorExitCode`

Reply:
Done